R/pin-delete.R

```diff
@@ -4,6 +4,8 @@
 #'
 #' @inheritParams pin_read
 #' @param names The names of one or more pins to delete
+#' @details
+#' Some boards require additional arguments when deleting pins. In particular, deleting pins in OneDrive and SharePoint Online (`board_onedrive`, `board_sharepoint` and `board_ms365`) may require passing `by_item=TRUE` to work around document protection policies that prohibit deleting non-empty folders.
```

Comment:
I think it's probably better to put this in the docs for `board_ms365()`?

Reply:
My thinking was, if I get an error on `pin_delete`, I'll most likely look in `?pin_delete` for help. Maybe put it in both places?

Reply:
`delete_by_item` is now an argument to `board_ms365`, so passing an arg to `pin_delete` is no longer necessary.